server/svix-server/src/error.rs

```diff
@@ -16,35 +16,53 @@ use serde_json::json;
 pub type Result<T> = std::result::Result<T, Error>;
 
 /// The error type returned from the Svix API
-#[derive(Debug, Clone)]
-pub enum Error {
-    /// A generic error
-    Generic(String),
-    /// Database error
-    Database(String),
-    /// Queue error
-    Queue(String),
-    /// Database error
-    Validation(String),
-    /// Any kind of HttpError
-    Http(HttpError),
+#[derive(Debug)]
+pub struct Error {
+    // the file name and line number of the error. Used for debugging non Http errors
+    pub location: &'static str,
+    pub error_type: ErrorType,
```

Comment:
Nitpick:
```suggestion
    pub type_: ErrorType,
```
or `t` or `typ`. The word error is redundant.